lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -534,6 +534,9 @@ static void diagnoseGeneralOverrideFailure(ValueDecl *decl,
                                            ArrayRef<OverrideMatch> matches,
                                            OverrideCheckingAttempt attempt) {
   auto &diags = decl->getASTContext().Diags;
+  if (cases where you have a property trying to override a no-args function on base class and vice-versa) {
```

Comment:
> @LucianoPAlmeida Here is my draft. I have no idea about how to detect expect cases.

If I understood correctly, what we have to check is:
* `decl` could not find a override match in parent class hierarchy. Which is already done in line 553, so I guess that is the block we the additional verification could be done.
* So if no override could be found for a `method` with no-args for example, we should verify is we could find an override by attempting **it as a property** (pretending this is a var `VarDecl`) and if there is a match override, provide a fix-it and the same vice-versa.
I can't suggest how to do in code because I don't know this part in detail, but I would suggest to look where this method is called and how this override matching is done, so from there we could provide the fix. 

Maybe @CodaFi, can give a more detailed and helpful guidance on the implementation